mlblocks/mlpipeline.py

```diff
@@ -777,7 +782,10 @@ def fit(self, X=None, y=None, output_=None, start_=None, debug=False, **kwargs):
 
             self._fit_block(block, block_name, context, debug_info)
 
-            if (block_name != self._last_block_name) or (block_name in output_blocks):
+            if block_name == self._last_fit_block:
+                early_stop = True
+
+            if (not early_stop) or (block_name in output_blocks):
```

Comment:
We also need to modify below, on line 798 (I cannot add a comment there because it has no changes):

The `if output_variables is not None and not output_blocks` can be changed to:

```python
if output_variables:
    if not output_blocks:
        # do the return logic here
elif not fit_pending:
    return
```

Reply:
thanks @csala for the review. I went ahead with the changes, I'm wondering if the return logic is a bit difficult to trace now
```python3
for block_name, block in self.blocks.items():
    # perform regular procedure

    if output_variables:
        if not output_blocks:
            # find result

            if debug:
                return result, debug_info

            return result

        elif not fit_pending:
            if debug:
                return debug_info

            return

# start_ check

if debug:
    return debug_info
```

Ideally, it's a lot easier for me to see what is happening when there are less `returns` to follow. I don't have a solution to this but I would prefer to `break` instead and let the last return carry it out.

```python3
if output_variables:
    if not output_blocks:
        # return logic
elif not fit_pending:
    break
```